pkg/ingester/lifecycle_test.go

```diff
@@ -20,14 +20,18 @@ import (
 	"github.com/grafana/dskit/ring"
 	"github.com/grafana/dskit/services"
 	"github.com/grafana/dskit/test"
+	"github.com/prometheus/common/model"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 
 	"github.com/grafana/mimir/pkg/ingester/client"
 	"github.com/grafana/mimir/pkg/util/validation"
 )
 
-const userID = "1"
+const (
+	userID              = "1"
+	testPastGracePeriod = 100 * 365 * 24 * time.Hour // 100 years, as many tests push samples around unix time 0 and we don't want to keep fixing this tests again and again.
```

Comment:
We can just disable it, right?

Reply:
I thought about using `0` value as `disabled`, but something inside of me didn't like it. I think it's fair to expect some kind of config set up.

Reply:
Yeah, that's what I was suggesting. We can set reasonable default (once it's tested), but having ability to disable would be nice.

Reply:
So you're suggesting to use zero value to disable it?

Reply:
After talking on Slack, I changed the behavior to interpret 0 as disabled.